### PR TITLE
More aws-s3 updates

### PR DESCRIPTION
**Pull request by vmo69**
Some improvements based on DOT teams feedback.

### CODE REVIEW

File: source/configuration_manual/mail_location/obox/amazon_s3.rst [hunk 1]
```diff
@@ -94,6 +85,33 @@ by adding the bucket region parameter to the S3 URL:
...
+parameters, refer to :ref:`http_based_object_storages` for details. The same
+setup can be achieved by using the ``s3`` scheme and adding the parameters
+manually.
+
+.. code-block:: none
+
+  addhdrvar=x-amz-security-token:%{auth:token}&loghdr=x-amz-request-id&loghdr=x-amz-id-2
+
+Configuring ``loghdr`` for ``x-amz-request-id`` and ``x-amz-id-2`` tells Dovecot to
+include these information in any error, debug or warning message. These
```
Comment (thuovila) [inline source/configuration_manual/mail_location/obox/amazon_s3.rst]: s/these/this information x 2 I think

File: source/configuration_manual/mail_location/obox/amazon_s3.rst [hunk 2]
```diff
@@ -94,6 +85,33 @@ by adding the bucket region parameter to the S3 URL:
...
+information help when Troubleshooting Amazon S3 (https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html).
```
Comment (thuovila) [inline source/configuration_manual/mail_location/obox/amazon_s3.rst]: This information helps

File: source/configuration_manual/mail_location/obox/amazon_s3.rst [hunk 3]
```diff
@@ -94,6 +85,33 @@ by adding the bucket region parameter to the S3 URL:
...
+You can also add other ``loghdr`` if needed.
```
Comment (thuovila) [inline source/configuration_manual/mail_location/obox/amazon_s3.rst]: "an other", I think, or is it specifically just "one other", or "one more"?

File: source/configuration_manual/mail_location/obox/amazon_s3.rst (hunk 1 above)
Comment (vmo69) [reply source/configuration_manual/mail_location/obox/amazon_s3.rst]: yes sounds better thanks

File: source/configuration_manual/mail_location/obox/amazon_s3.rst [hunk 4]
```diff
@@ -94,6 +85,33 @@ by adding the bucket region parameter to the S3 URL:
...
+You can also add multiple other ``loghdr`` parameters if needed.
+
+Debug Output Example:
+
+.. code-block:: none
+
+        doveadm(user): Debug: http-client: conn 1.2.3.4:443 [1]: Got 200 response for request [Req1: GET https://test-mails.s3-service.com/?prefix=user%2Fidx%2F]: OK (x-amz-request-id:AABBCC22BB7798869, x-amz-id-2:DeadBeefanXBapRucWGAD1+aWwYMfwmXydlI0mHSuh4ic/j8Ji7gicTsP7xpMQz1IR9eydzeVI=) (took 63 ms + 140 ms in queue)
+
+The ``addhdrvar`` parameter with ``x-amz-security-token`` adds the ``%{auth:token}``
+expanded variables value to all HTTP requests if the value is defined. This
```
Comment (sirainen) [inline source/configuration_manual/mail_location/obox/amazon_s3.rst]: variable's

File: source/configuration_manual/mail_location/obox/amazon_s3.rst [hunk 5]
```diff
@@ -94,6 +85,33 @@ by adding the bucket region parameter to the S3 URL:
...
+variable does get set by Dovecot internally in case AWS IAM is in use.
```
Comment (sirainen) [inline source/configuration_manual/mail_location/obox/amazon_s3.rst]: s/does get/gets/

File: source/configuration_manual/mail_location/obox/amazon_s3.rst (hunk 4 above)
Comment (vmo69) [reply source/configuration_manual/mail_location/obox/amazon_s3.rst]: hmm, i never get this right thanks.

**Pull request merged**